walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList; 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;   
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    /**
+    * Returns a collection of available TimeRange given a collection of Event objects and
+    * MeetingRequest. A TimeRange is available if the person in MeetingRequest are all available.
+    * We first do this by checking which TimeRange are the attendees not available in and 
+    * use sliding window to record the available TimeRange provided they satisfy the duration.
+    * @param    events  a collection of Event objects
+    * @param    request a MeetingRequest object
+    * @return           collection of available TimeRange  
+    */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        ArrayList<TimeRange> overlappingTime = new ArrayList<TimeRange>();
+        ArrayList<TimeRange> mergedEventTime = new ArrayList<TimeRange>();
+        ArrayList<TimeRange> availableTime = new ArrayList<TimeRange>();
+        Set<String> requestAttendees = new HashSet<String>(request.getAttendees());
+
+        // record TimeRange whereby any of attendees is unavailable, i.e. in another event
+        for(Event event: events){
+            Set<String> eventAttendees = new HashSet<String>(event.getAttendees());
+            eventAttendees.retainAll(requestAttendees);
+            if (!eventAttendees.isEmpty()) {
+                overlappingTime.add(event.getWhen());
+            }
+        }
+
+        // merge eventTime time range
+        mergedEventTime = mergeEventTime(overlappingTime);
+
+        // sliding window to find available timing that matches the duration
+        int prevEnd = TimeRange.START_OF_DAY;
+        long duration=request.getDuration();
```

Comment:
Minor style inconsistency: leave one space before and after '='. That's basically the same for all binary operators.

Reply:
ah i missed that. what would one do to get a linter :)